File: Codings/Cpp/08_MineSweeper.cpp
std::vector<std::vector<int>> minesweeper(std::vector<std::vector<bool>> matrix) {
    vector < vector <int> > r (matrix.size(), vector <int> (matrix[0].size()));
    for(int i = 0; i < matrix.size(); i++) {
        for(int j = 0; j < matrix[0].size(); j++) {
            r[i][j] += i > 0 && j > 0 && matrix[i-1][j-1];
            r[i][j] += i > 0 && matrix[i-1][j];
            r[i][j] += i > 0 && j < matrix[0].size() - 1 && matrix[i-1][j+1];
            r[i][j] += j > 0 && matrix[i][j-1];
            r[i][j] += j < matrix[0].size() - 1 && matrix[i][j+1];
            r[i][j] += i < matrix.size() - 1 && j > 0 && matrix[i+1][j-1];
            r[i][j] += i < matrix.size() - 1 && matrix[i+1][j];
            r[i][j] += i < matrix.size() - 1 && j < matrix[0].size() - 1 && matrix[i+1][j+1];
        }
    }
    return r;
}

File: Codings/Java/08MineSweeper.java
int[][] minesweeper(boolean[][] matrix) {
        int[][] answer = new int[matrix.length][matrix[0].length];
        for(int rows = 0; rows < matrix.length; rows++)
        {
            for(int columns = 0; columns < matrix[0].length; columns++)
            {
                answer[rows][columns] = convertedFieldValue(matrix, rows, columns);
            }
        }
        return answer;
}


 int convertedFieldValue(boolean[][] matrix, int row, int column) {
    int sum =
            boolToInt(matrix, row - 1, column - 1) +
            boolToInt(matrix, row - 1, column) +
            boolToInt(matrix, row - 1, column + 1) +
            boolToInt(matrix, row, column - 1) +
            boolToInt(matrix, row, column + 1) +
            boolToInt(matrix, row + 1, column - 1) +
            boolToInt(matrix, row + 1, column) +
            boolToInt(matrix, row + 1, column + 1);
     return sum;
}

  int boolToInt(boolean[][] matrix, int row, int column) {
    try {
        if (!matrix[row][column]) return 0;
        else return 1;
    } catch (ArrayIndexOutOfBoundsException e) {
        return 0;
    } catch (Exception e) {
        System.out.println("Something went wrong.");
        return 0;
    }
}
